subsystems/viamserver/viamserver.go

```diff
@@ -135,7 +154,7 @@ func (s *viamServer) Start(ctx context.Context) error {
 		return nil
 	case <-ctx.Done():
 		return ctx.Err()
-	case <-time.After(startTimeout):
+	case <-time.After(s.startTimeout()):
```

Comment:
It'd be better to parse all the necessary fields out of the config into a new struct and use those directly. Easier for logging and debugging.
This may be fine for a one-off but this pattern won't scale. Is this a config-reading pattern we use in the rest of the code base?

Reply:
switch to something more general in 9155ca4

in the codebase as-is we use go-flags for env/cli flags, other than that just proto and json afaik

the golang config library I'm most familiar with is viper -- has defaults + duration parsing, would probably make this PR a purely declarative change. (though our config approach has three layers and may not fit perfectly in a third party library)